Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -2637,9 +2646,63 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     }
 }
 
+WKContentRuleListStore *WebExtensionContext::declarativeNetRequestRuleStore()
+{
+    String contentBlockerStorePath = m_extensionController->configuration().declarativeNetRequestStoreDirectory();
+    if (!FileSystem::makeAllDirectories(contentBlockerStorePath))
+        return nil;
+
+    return [WKContentRuleListStore storeWithURL:[NSURL fileURLWithPath:contentBlockerStorePath]];
```

Comment:
This makes a new `WKContentRuleListStore` each time, should it be cached on the context? We also likely only want to try to make the directories in the path the first time its needed.

Reply:
Agreed. I'll cache it.